### PR TITLE
enh(clapi): handle login throttling and password expiration in CLAPI

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to handle login throttling and password expiration in CLAPI

**Fixes** # MON-6472

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/centreon-clapi/centreonAPI.class.php [hunk 1]
```diff
@@ -1148,4 +1207,66 @@ public function sortClassExport()
...
+    {
+        $blockLoginStatement = $this->DB->prepare(
+            'UPDATE contact SET blocking_time = :blockingTime WHERE contact_alias = :contactAlias'
+        );
+        $blockLoginStatement->bindValue(':blockingTime', time(), \PDO::PARAM_INT);
+        $blockLoginStatement->bindValue(':contactAlias', $this->login, \PDO::PARAM_STR);
+        $blockLoginStatement->execute();
+    }
+
+    private function exitOnInvalidCredentials(
```
Comment (AssanDiomande) [inline www/class/centreon-clapi/centreonAPI.class.php:1246]: missing phpdoc

File: www/class/centreon-clapi/centreonAPI.class.php [hunk 2]
```diff
@@ -520,6 +524,54 @@ public function checkUser($useSha1 = false, $isWorker = false)
                 exit(1);
             }
 
+            // Get Security Policy
+            $statement = $this->DB->prepare(
+                'SELECT `custom_configuration` FROM provider_configuration WHERE name = :name'
```
Comment (kduret) [inline www/class/centreon-clapi/centreonAPI.class.php:529]: use `getPasswordSecurityPolicy` from `www/class/centreonContact.class.php`

File: www/class/centreon-clapi/centreonAPI.class.php [hunk 3]
```diff
@@ -519,6 +523,42 @@ public function checkUser($useSha1 = false, $isWorker = false)
                 print "You don't have permissions for CLAPI.\n";
                 exit(1);
             }
+            $contact = new \CentreonContact($this->DB);
```
Comment (kduret) [inline www/class/centreon-clapi/centreonAPI.class.php:526]: remove contact instanciation L573

File: www/class/centreon-clapi/centreonAPI.class.php [hunk 4]
```diff
@@ -519,6 +523,42 @@ public function checkUser($useSha1 = false, $isWorker = false)
...
+
+            // Check if user is blocked
+            if ($row['blocking_time'] !== null) {
+                // If he is block and blocking duration is expired, unblock him
+                if ((int) $row['blocking_time'] + (int) $securityPolicy['blocking_duration'] < time()) {
+                    $this->removeBlockingTimeOnUser();
+                    $row['login_attempts'] = null;
+                    $row['blocking_time'] = null;
+                } else {
+                    $now = (new \DateTime())->setTimestamp(time());
```
Comment (kduret) [inline www/class/centreon-clapi/centreonAPI.class.php:545]:
```suggestion
                    $now = new \DateTime()
```

**Review by kduret — APPROVED** [review]

**Review by AssanDiomande — APPROVED** [review]

**Pull request merged**